services/swagger/swagger.service.js

```diff
@@ -22,20 +22,21 @@ module.exports = class SwaggerValidatorService extends BaseJsonService {
   static get route() {
     return {
       base: 'swagger/valid/2.0',
-      pattern: ':scheme(http|https)?/:url*',
+      pattern: ':scheme(http|https)?/:fileExtension(json|yaml)?/:url*',
```

Comment:
We should move the spec url path from route parameters to a single query param, with a redirector to maintain backwards compatibility for existing badges. 

The Website service is probably a decent reference, https://github.com/badges/shields/pull/4028/files

https://github.com/badges/shields/blob/7f54e09fe542578f8d4488aab9d020b54d0f0e8c/services/website/website.service.js#L41-L47
https://github.com/badges/shields/blob/master/services/website/website-redirect.service.js

Reply:
Actually the TwitterUrl badge may be a simpler/better reference, as neither it nor this Swagger service had the same complexity the Website badge did 

https://github.com/badges/shields/blob/master/services/twitter/twitter.service.js
https://github.com/badges/shields/blob/master/services/twitter/twitter-redirect.service.js

Reply:
I managed to move it to a query param, will look into the redirector in the following week

Reply:
So I wrote the redirector, and theoretically its working, but I've got one problem
`module.exports = [
  redirector({
    category: 'other',
    name: 'SwaggerUrlRedirect',
    route: {
      base: 'swagger/valid/2.0',
      pattern: ':scheme(http|https)?/:url*',
    },
    transformPath: () => `/swagger/valid/3.0/spec`,
    transformQueryParams: ({ scheme, url }) => ({
      url: `${scheme}://${url}`
    }),
    dateAdded: new Date('2019-11-03'),
  }),
]`
This
`http://localhost:8080/swagger/valid/2.0/https/raw.githubusercontent.com%2FOAI%2FOpenAPI-Specification%2Fmaster%2Fexamples%2Fv2.0%2Fjson%2Fp.etstore-expanded.json`
will get
`http://localhost:8080/swagger/valid/3.0/spec.json?url=https%3A%2F%2Fraw.githubusercontent.com%2FOAI%2FOpenAPI-Specification%2Fmaster%2Fexamples%2Fv2.0%2Fjson%2Fp.etstore-expanded`

So the file extension does get appended on my path instead of begin part of the pattern. Any idea why?

Reply:
For the route `pattern`, try this instead:

`pattern: ':scheme(http|https)?/:url+',`

Also, I don't think the `scheme` portion should really be optional (I know it was already, just think we should also change that too given what we are issuing to the upstream API)

https://github.com/badges/shields/blob/7f54e09fe542578f8d4488aab9d020b54d0f0e8c/services/swagger/swagger.service.js#L52-L63

Reply:
Unfortunately the changed pattern does not do any difference. Its just with json, a yaml file ending does get translated correctly

Reply:
Did you also remove the `?` from the scheme?

Reply:
Tried with removing and without yes

Reply:
Ah okay, would you mind trying `:url(.+)` as well? Sorry I know I've seen this before recently just don't remember off hand what the fix was

Reply:
No change also tried with * and (:url.+) and :url(.+)+

Reply:
Could this be some sort of issue with the badge formats? Supported is svg and json ... and the .json is interpreted as the format of the badge when I see correctly